src/emulator/modules/SceGxm/SceGxm.cpp

```diff
@@ -719,11 +736,60 @@ EXPORT(int, sceGxmGetRenderTargetMemSizes) {
     return unimplemented("sceGxmGetRenderTargetMemSizes");
 }
 
+struct GxmThreadParams {
+    KernelState *kernel = nullptr;
+    MemState *mem = nullptr;
+    SceUID thid = SCE_KERNEL_ERROR_ILLEGAL_THREAD_ID;
+    GxmState *gxm = nullptr;
+};
+
+static int SDLCALL thread_function(void *data) {
+    const GxmThreadParams params = *static_cast<const GxmThreadParams *>(data);
+    while(true){
```

Comment:
There's a chance here that the parameters passed to the thread function are destroyed before the thread properly starts. When creating this thread, we need to wait for the thread to indicate that it's made a copy of the arguments before the original goes out of scope.

Search for `host_may_destroy_params` to see how it's done.